markdown/events/2021/bytesize-4-github-contribution-basics.md

```diff
@@ -34,3 +34,96 @@ This will cover:
+
+Another best practice in code review, I should start with first talking about code reviews in general. So what we do, and of course whenever somebody opens a pull request, then we check this code against the rules that the nf-core has specified. So for example whether they are in the same code style that the markdown documentation is in line with. What we typically adhere to, is that the Nextflow code is written in a way not known to produce any bugs that we are aware of. The general rule is and you can also check that in the guidelines that I just spoke about.
+
+Pull requests to the dev branch, so to the experimental code, requires one review by someone in the nf-core community, whereas the pull requests to the master branch actually require two independent reviewers. That's something a lot of people struggle with, they don't know about these rules, although these are I think written down somewhere, hidden deep within these guidelines actually. There are also some nice how-tos on how code can be reviewed effectively. [19:30](https://youtu.be/gTEXDXWf4hE?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=1115)
+
+An example that I've found particularly helpful is actually listed on the [slides](https://doi.org/10.6084/m9.figshare.14160680.v1). It's free, and most of the rules that are written in there are actually also applied in the same way here in nf core when we do a code reviews. [19:34](https://youtu.be/gTEXDXWf4hE?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=1174)
+
+One of the important steps in code review is that you should read things quite carefully, so for example, a lot of people write code at night, and are very happy if people catch typos and things like that. So you need to carefully read documentation, and it's very good to use certain features in Github. [20:01](https://youtu.be/gTEXDXWf4hE?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=1201)
+
+So for example, there is a new feature; whenever there's a pull request open you can click on files changed and you can go through the files that have been changed in that pull request and click on insert a suggestion. [20:11](https://youtu.be/gTEXDXWf4hE?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=1211)
+
+Whenever you click on the line of the code that would then open a text box, they can actually suggest a change that would make it possible later on for the person who opened the pull request to just click on accept that change. [20:17](https://youtu.be/gTEXDXWf4hE?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=1217)
+
+For example for this type here it would be very easy to just type here it would be very easy to just such a suggestion fixing, and people don't have to write their own code again or go back to the editor because they can just click on this in the github interface and amend things. [20:46](https://youtu.be/gTEXDXWf4hE?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=1246)
```

Comment:
```suggestion
For example for this type here it would be very easy to just type here and add such a suggestion fixing, and people don't have to write their own code again or go back to the editor because they can just click on this in the github interface and amend things. [20:46](https://youtu.be/gTEXDXWf4hE?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=1246)
```